mercury_agent/hardware/drivers/megaraid.py

```diff
@@ -69,14 +82,232 @@ def get_vendor_info(adapter):
             'supported_adapter_ops': adapter['Supported Adapter Operations'],
             'supported_pd_ops': adapter['Supported PD Operations'],
             'supported_vd_ops': adapter['Supported VD Operations'],
-            'bbu_info': adapter['BBU_Info']
+            'bbu_info': adapter.get('BBU_Info')
+        }
+
+    @staticmethod
+    def dash_is_none(value):
+        """
+
+        :param value:
+        :return:
+        """
+        if value != '-':
+            return value
+
+    def populate_logical_drives(self, disk_group, dg_info):
+        """
+
+        :param disk_group:
+        :param dg_info:
+        :return:
+        """
+        lds = []
+        for vd in dg_info['VD LIST']:
+            dg_id, vd_id = (int(x) for x in vd['DG/VD'].split('/'))
+            if dg_id == disk_group:
+                lds.append({
+                    'level': int(vd['TYPE'].strip('RAID')),
+                    'size': Size(vd['Size'], force_iec_values=True).bytes,
+                    'status': self.logical_drive_map.get((vd['State']),
+                                                         'UNKNOWN'),
+                    'extra': {
+                        'disk_group': dg_id,
+                        'access': vd['Access'],
+                        'consistency': vd['Consist'],
+                        'cache': vd['Cache'],
+                        'name': vd['Name'],
+                        'vendor_state': vd['State']
+                    }
+                })
+
+        return lds
+
+    def format_physical_drive(self, pd):
+        """
+
+        :param pd:
+        :return:
+        """
+        # DG JSON type is inconsistent
+        disk_group = self.dash_is_none(pd['DG'])
+        if isinstance(disk_group, str):
+            disk_group = int(disk_group)
+
+        return {
+            'size': Size(pd['Size'], force_iec_values=True).bytes,
+            'status': self.drive_status_map.get(pd['State'], 'UNKNOWN'),
+            'type': pd['Med'],
+            'extra': {
+                'address': pd['EID:Slt'],
+                'drive_id': pd['DID'],
+                'model': pd['Model'],
+                'disk_group': disk_group,
+                'interface': pd['Intf'],
+                'self_encrypted_drive':
+                    pd['SED'] != 'N' and True or False,
```

Comment:
Would prefer a more explicit 'lowbrow' logic here

Reply:
what would you suggest?

Reply:
Why not just `pd['SED'] != 'N'?`

Reply:
Yes, that is better :1st_place_medal: